From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: microsoft-golang-bot <microsoft-golang-bot@users.noreply.github.com>
Date: Fri, 11 Feb 2022 18:03:46 -0600
Subject: [PATCH] Check signature of microsoft/go builds

Change ".asc" to ".sig" to match the filename we publish. Download our
public key from an https Microsoft URL to check the signature against.
---
 Dockerfile-linux.template | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Dockerfile-linux.template b/Dockerfile-linux.template
index cde43a9..c300098 100644
--- a/Dockerfile-linux.template
+++ b/Dockerfile-linux.template
@@ -109,12 +109,14 @@ RUN set -eux; \
 {{ ) end -}}
 	fi; \
 	\
-	wget -O go.tgz.asc "$url.asc"; \
+	wget -O go.tgz.asc "$url.sig"; \
 	wget -O go.tgz "$url"{{ if is_alpine then "" else " --progress=dot:giga" end }}; \
 	echo "$sha256 *go.tgz" | sha256sum -c -; \
 	\
 # https://github.com/golang/go/issues/14739#issuecomment-324767697
 	GNUPGHOME="$(mktemp -d)"; export GNUPGHOME; \
+	wget -O microsoft-managed-lang-compiler.asc 'https://download.microsoft.com/download/f/a/2/fa2420dd-3f08-4621-82cf-5a22abcbc8f9/microsoft-managed-lang-compiler.asc'; \
+	gpg --batch --import microsoft-managed-lang-compiler.asc; \
 # https://www.google.com/linuxrepositories/
 	gpg --batch --keyserver keyserver.ubuntu.com --recv-keys 'EB4C 1BFD 4F04 2F6D DDCC  EC91 7721 F63B D38B 4796'; \
 # let's also fetch the specific subkey of that key explicitly that we expect "go.tgz.asc" to be signed by, just to make sure we definitely have it
